lib/internal/modules/esm/resolve.js

```diff
@@ -109,19 +111,22 @@ function getPackageConfig(path) {
     throw new ERR_INVALID_PACKAGE_CONFIG(errorPath, error.message, true);
   }
 
-  let { main, name, type } = packageJSON;
+  let { imports, main, name, type } = packageJSON;
   const { exports } = packageJSON;
+  if (typeof imports !== 'object' || imports === null) imports = undefined;
   if (typeof main !== 'string') main = undefined;
   if (typeof name !== 'string') name = undefined;
   // Ignore unknown types for forwards compatibility
   if (type !== 'module' && type !== 'commonjs') type = 'none';
 
   const packageConfig = {
+    pjsonPath: path,
     exists: true,
     main,
     name,
     type,
-    exports
+    exports,
+    imports
```

Comment:
nit: trailing comma?

```suggestion
    imports,
```